htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/Table.java

```diff
@@ -49,68 +45,70 @@ public Table(WebElement wrappedElement) {
      * @return List with text values of table heading elements.
      */
     public List<String> getHeadingsAsString() {
-        return convert(getHeadings(), toTextValues());
+        return getHeadings().stream()
+                .map(WebElement::getText)
+                .collect(toList());
     }
 
     /**
-     * Returns table cell elements grouped by rows.
+     * Returns table cell elements ({@code <td>}), grouped by rows.
      *
      * @return List where each item is a table row.
      */
     public List<List<WebElement>> getRows() {
-        List<List<WebElement>> rows = new ArrayList<>();
-        List<WebElement> rowElements = getWrappedElement().findElements(By.xpath(".//tr"));
-        for (WebElement rowElement : rowElements) {
-            rows.add(rowElement.findElements(By.xpath(".//td")));
-        }
-        return rows;
+        return getWrappedElement()
+                .findElements(By.cssSelector("tr"))
+                .stream()
+                .map(rowElement -> rowElement.findElements(By.cssSelector("td")))
+                .filter(row -> row.size() > 0) // ignore rows with no <td> tags
+                .collect(toList());
     }
 
     /**
-     * Returns text values of table cell elements grouped by rows.
+     * Returns text values of table cell elements ({@code <td>}), grouped by rows.
      *
      * @return List where each item is text values of a table row.
      */
     public List<List<String>> getRowsAsString() {
-        return convert(getRows(), toListsConvertingEachItem(toTextValues()));
+        return getRows().stream()
```

Comment:
you can then getValues as list and it will be same result. But think its ok to leave it as you done